File: frontend/src/services/userService.js
import api from './api';

export const getUserProfile = async (userId) => {
    const response = await api.get(`/users/${userId}`);
    return response.data;
};

export const updateUserPreferences = async (userId, studyPreferences) => {
    const response = await api.put(`/users/${userId}/preferences`, { studyPreferences });
    return response.data;
};

export const getUserReviews = async (userId) => {
    const response = await api.get(`/users/${userId}/reviews`);
    return response.data;
};

File: backend/src/main/java/com/wheretostudy/backend/service/UserService.java
package com.wheretostudy.backend.service;

import com.wheretostudy.backend.dto.UserProfileDTO;
import com.wheretostudy.backend.model.User;
import com.wheretostudy.backend.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    public UserProfileDTO getUserProfile(UUID userId) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("User not found"));
        return toUserProfileDTO(user);
    }

    public UserProfileDTO updateUserPreferences(UUID userId, String preferences) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("User not found"));
        
        user.setStudyPreferences(preferences);
        User savedUser = userRepository.save(user);
        
        return toUserProfileDTO(savedUser);
    }

    private UserProfileDTO toUserProfileDTO(User user) {
        return UserProfileDTO.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .role(user.getRole())
                .studyPreferences(user.getStudyPreferences())
                .build();
    }
}
